core/src/apps/cardano/sign_tx/ordinary_signer.py

```diff
@@ -30,6 +30,9 @@ def _validate_tx_init(self) -> None:
         if (
```

Comment:
Looking at these conditions now, I'd consider adding a `_assert_tx_init_cond` (or similar name) to validate tx init in the signers so that when reading the conditions it would be apparent what needs to be fulfilled. Currently when reading the code I need to always reverse the conditions in my head 🤷 E.g.:
```
self._assert_tx_init_cond(self.msg.collateral_inputs_count == 0)
self._assert_tx_init_cond(self.msg.required_signers_count == 0)
self._assert_tx_init_cond(self.msg.has_collateral_return == False)
self._assert_tx_init_cond(self.msg.total_collateral is None)
self._assert_tx_init_cond(self.msg.reference_inputs_count == 0)
```

In `PoolOwnerSigner` we could IMHO merge the two "condition groups" into `Invalid tx signing request`.

Reply:
I agree.